Sources/HaCTML/AttributeMap.swift

```diff
@@ -0,0 +1,42 @@
+/**
+  A collection of attributes for a single HTML element
+ */
+struct AttributeMap: ExpressibleByArrayLiteral {
+  typealias Element = Attribute
+
+  // TODO: Make the key for this AttributeKey, so we can have multiple AttributeKeys with the same name
+  private let attributeMap: Dictionary<String, Attribute>
+
+  init(attributes arrayLiteral: [Element]) {
+    var newAttributes = Dictionary<String, Attribute>()
+
+    for attribute in arrayLiteral {
+      newAttributes[attribute.key.keyName] = attribute
+    }
+
+    attributeMap = newAttributes
+  }
+  
+  init (arrayLiteral: Element...) {
+    self.init(attributes: arrayLiteral)
+  }
+
+  func get<Value>(key: AttributeKey<Value>) -> Value? {
+    // TODO: Implement this as a subscript once we move to Swift 4
+    //   It can't be one now as generic subscripts aren't allowed prior to Swift 4,
+    //   and this function is generic
+    return attributeMap[key.keyName]?.value as! Value?
```

Comment:
This _is_ safer as it catches the case when our map contains something it shouldn't. Otherwise it would silently return `nil`.

Reply:
It's not necessarily an invalid dictionary. If somebody defines their own Attribute for some reason which happens to have the same name as one of the defaults but a different Value type, this could crash where it should be expected to return nil.

Reply:
In your explained scenario I'd prefer it to crash, though there's a `TODO` to fix this so the key of the dictionary are the attribute key objects, rather than their string keys.

Reply:
Probably safe to leave this discussion for when we do the Swift 4 move, am happy for this to be in there for now